conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Parameters.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.collect.Multimap;
+import com.palantir.logsafe.SafeArg;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.AttachmentKey;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class to put different types of safe or unsafe parameters in a multimap attached the exchange.
+ * It requires the exchanged o have been previously through a ParametersHandler which would have created the attached
+ * multimaps..
+ */
+public final class Parameters {
+
+    private static final Logger log = LoggerFactory.getLogger(Parameters.class);
+
+    public static final AttachmentKey<Multimap<String, Object>> SAFE_PARAMS_ATTACH_KEY =
+            AttachmentKey.create(Multimap.class);
+    public static final AttachmentKey<Multimap<String, Object>> UNSAFE_PARAMS_ATTACH_KEY =
+            AttachmentKey.create(Multimap.class);
+
+    private Parameters() {}
+
+    public static void putSafePathParam(HttpServerExchange exchange, String key, Object value) {
+        putParam(exchange, SAFE_PARAMS_ATTACH_KEY, key, value);
+    }
+
+    public static void putUnsafePathParam(HttpServerExchange exchange, String key, Object value) {
+        putParam(exchange, UNSAFE_PARAMS_ATTACH_KEY, key, value);
+    }
+
+    public static void putSafeQueryParam(HttpServerExchange exchange, String key, Object value) {
```

Comment:
collapse to putSafeParam and putUnsafeParam

Reply:
On that one I was less certain cause I felt that this is an internal detail. If we ever decide to revert that decision and split the different types, the generated code is still valid and we only need to change the lib. What do you think?

Reply:
This is the last point to resolve. Do you feel strongly about this?